### PR TITLE
Modification of compilation units generation

### PR DESCRIPTION
**Pull request by bitmouse**
Modified way of generating unity files to keep same content and timestamp
as long as possible to avoid recompilation of units.
New approach generates only units if new files are added
or files are removed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -96,8 +97,10 @@ facelift_export_project(
         cmake/faceliftMacros.cmake
         ${CMAKE_BINARY_DIR}/${PROJECT_NAME}Variables.cmake
         ${CMAKE_BINARY_DIR}/faceliftConfig-codegen.cmake
+        ${CMAKE_BINARY_DIR}/unityConfig-facelift.cmake
     INSTALLED_FILES
         cmake/faceliftMacros.cmake
         ${CMAKE_BINARY_DIR}/${PROJECT_NAME}Variables.cmake.installed
         ${CMAKE_BINARY_DIR}/faceliftConfig-codegen.installed.cmake
+        ${CMAKE_BINARY_DIR}/unityConfig-facelift.installed.cmake
```
Comment (jacky309) [inline CMakeLists.txt]: faceliftUnityConfig.installed.cmake

File: CMakeLists.txt [hunk 2]
```diff
@@ -96,8 +97,10 @@ facelift_export_project(
         cmake/faceliftMacros.cmake
         ${CMAKE_BINARY_DIR}/${PROJECT_NAME}Variables.cmake
         ${CMAKE_BINARY_DIR}/faceliftConfig-codegen.cmake
+        ${CMAKE_BINARY_DIR}/unityConfig-facelift.cmake
```
Comment (jacky309) [inline CMakeLists.txt]: same as below

File: CMakeLists.txt (hunk 1 above)
Comment (bitmouse) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 2 above)
Comment (bitmouse) [reply CMakeLists.txt]: fixed

File: unity/faceliftUnityConfig.cmake.in [hunk 3]
```diff
@@ -0,0 +1,4 @@
+if (NOT TARGET unity_generator)
```
Comment (jacky309) [inline unity/faceliftUnityConfig.cmake.in]: move the content of the "unity" folder into the existing "cmake" folder.

**Pull request merged**